int8 state_green_walk
int8 state_random_walk
int8 state_random_turn
int8 finished_action